### PR TITLE
Simplify ModelController

### PR DESCRIPTION
**Pull request by Corvince**
Not the most urgent of things to do, but I was in the mood to play around with this. Overall, should be much simpler now.

### CODE REVIEW

**Review by EwoutH — APPROVED** [review]
I have no idea what any of this code does, but I always like simpler!

File: mesa/visualization/solara_viz.py [hunk 1]
```diff
@@ -154,110 +153,49 @@ def ModelController(model: solara.Reactive["Model"], play_interval):
...
-            playing.value = False
-        elif model.value.running:
+    def step():
+        while playing.value:
+            time.sleep(play_interval / 1000)
             do_step()
-        else:
-            playing.value = False
+
+    solara.use_thread(step, [playing.value])
```
Comment (rht) [inline mesa/visualization/solara_viz.py:175]: Needs to test if this works on Google Colab.

**Comment by rht** [comment]
It is indeed simpler, more readable, and hence can be extended by users if they need to fork the component. The only concern is if it works on Google Colab.

**Comment by Corvince** [comment]
Confirmed that it works in Google Colab!

**Pull request merged**